src/Lib/Managers/NobManager.vala

```diff
@@ -74,348 +80,433 @@ public class Akira.Lib.Managers.NobManager : Object {
-        on_add_select_effect (canvas.selected_bound_manager.selected_items);
-    }
-
+    /*
+     * Set which nob is selected by its Nob name.
+     */
     public void set_selected_by_name (Nob selected_nob) {
         this.selected_nob = selected_nob;
     }
 
+    /*
+     * Compares a target item to the current nobs to see if there is a match. Otherwise returns Nob.NONE.
+     */
     public Nob get_grabbed_id (Goo.CanvasItem? target) {
-        int grabbed_id = -1;
+        int grabbed_id = Nob.NONE;
 
-        for (int i = 0; i < 9; i++) {
-            if (target == nobs[i]) grabbed_id = i;
+        for (var i = 0; i < 9; ++i) {
+            var nob = nobs[i];
+            if (target == nob) {
+                grabbed_id = i;
+                break;
+            }
         }
 
         return (Nob) grabbed_id;
     }
 
     public static bool is_top_nob (Nob nob) {
-        return nob == 0 || nob == 1 || nob == 2;
+        return nob == Nob.TOP_LEFT || nob == Nob.TOP_CENTER || nob == Nob.TOP_RIGHT;
     }
 
     public static bool is_bot_nob (Nob nob) {
-        return nob == 6 || nob == 5 || nob == 4;
+        return nob == Nob.BOTTOM_LEFT || nob == Nob.BOTTOM_CENTER || nob == Nob.BOTTOM_RIGHT;
     }
 
     public static bool is_left_nob (Nob nob) {
-        return nob == 0 || nob == 7 || nob == 6;
+        return nob == Nob.TOP_LEFT || nob == Nob.LEFT_CENTER || nob == Nob.BOTTOM_LEFT;
     }
 
     public static bool is_right_nob (Nob nob) {
-        return nob == 2 || nob == 3 || nob == 4;
+        return nob == Nob.TOP_RIGHT || nob == Nob.RIGHT_CENTER || nob == Nob.BOTTOM_RIGHT;
     }
 
-    private void update_select_bb_coords (List<Items.CanvasItem> selected_items) {
-        // Bounding box edges
-        double bb_left = 1e6, bb_top = 1e6, bb_right = 0, bb_bottom = 0;
+    /*
+     * Takes a set of items and populates information needed to determine the selection box and nob positions.
+     * If the number of items is one, the selection box may be rotated, otherwise it is never rotated.
+     */
+    public static void populate_nob_bounds_from_items (
+        List<Items.CanvasItem> items,
+        ref Cairo.Matrix matrix,
+        ref double top_left_x,
+        ref double top_left_y,
+        ref double width_offset_x,
+        ref double width_offset_y,
+        ref double height_offset_x,
+        ref double height_offset_y,
+        ref double width,
+        ref double height
+    ) {
+        top_left_x = 0;
+        top_left_y = 0;
 
-        foreach (var item in selected_items) {
-            bb_left = double.min (bb_left, item.coordinates.x1);
-            bb_top = double.min (bb_top, item.coordinates.y1);
-            bb_right = double.max (bb_right, item.coordinates.x2);
-            bb_bottom = double.max (bb_bottom, item.coordinates.y2);
+        if (items.length () == 1) {
+            var item = items.first ().data;
+            item.get_transform (out matrix);
+
+            Cairo.Matrix nob_matrix = matrix;
```

Comment:
Since we're doing an early return, we don't need to declare the `nob_matrix` and we can use the existing. `matrix` variable.